src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -752,11 +752,34 @@ def assign_college_simulants_guaridans(self, pop: pd.DataFrame) -> pd.DataFrame:
                     households_guardian_types[guardian_type]["race_ethnicity"] == race,
                     "reference_person_id",
                 ]
-                race_guardian_ids = self.randomness.choice(
-                    race_college_sims_with_guardian_idx,
-                    choices=race_reference_person_ids,
-                    additional_key=f"{race}_{guardian_type}_guardian_ids",
-                )
+                if (
+                    race_reference_person_ids.empty & len(race_college_sims_with_guardian_idx)
+                    > 0
+                ):
+                    # Get ids of for all other race reference persons if guardian type has no reference persons
+                    # This handles an unlikely edge case
+                    race_reference_person_ids = pd.concat(
+                        [
+                            df
+                            for gt, df in households_guardian_types.items()
+                            if gt != guardian_type
+                        ]
+                    )
+                    race_reference_person_ids = race_reference_person_ids.loc[
+                        race_reference_person_ids["race_ethnicity"] == race,
+                        "reference_person_id",
+                    ]
+                    race_guardian_ids = self.randomness.choice(
```

Comment:
Consider removing this chunk from the if statement and instead generating the additional_key. Then in the else block below just assign the different additional_key.

Then you need this chunk only once outside of the if/else

Reply:
Ok this makes it a lot cleaner and removes the duplicated chunk of code and I just left the additional key as guardian_type since that was more clear to me anyways.